src/main/Java/SampleApp.java

```diff
@@ -0,0 +1,267 @@
+			System.out.println("Saved product information to products.csv");
+		}
+		
+		/**
+		 * Example 3: 
+		 * a. call the API /search to get all Certified Product Details
+		 * b. Save results to a .txt file
+		 */
+		if(input == 3 || input == 0){
+			sampleApp.getSearchResult_saveResultsInFile();
+		}
+		
+		/**
+		 * Example 4:
+		 * a. Call the API /data/search_options using token for authentication
+		 */
+		if(input == 4 || input == 0){
+			sampleApp.getSearchOptions_usesAuthentication(token);
+		}
+		System.out.println("Completed execution. Exiting application...");
+	}
+	
+	/**
+	 * Displays instructions on how to use the CHPL Sample Application
+	 */
+	public void displayInstructions(){
+		System.out.println("Welcome to the CHPL Sample Application! The purpose of this application is to provide the following examples: \n");
+		
+		System.out.println("Example 1: ");
+		System.out.println("a. Call the API /auth/authentication using HTTP POST to get a valid token");
+		System.out.println("(note: the authentication token is not necessary for many API calls, including /search)");
+		System.out.println("b. Parse JSON response using com.google.gson");
+		System.out.println("c. Return token as string\n");
+		
+		System.out.println("Example 2: ");
+		System.out.println("a. Call the API /search using HTTP GET to get all Certified Product Details");
+		System.out.println("b. Parse the JSON using org.json to get only information keyed on \"product\"");
+		System.out.println("c. Parse the \"product\" JSON to CSV format");
+		System.out.println("d. Write the csv data to a file\n");
+		
+		System.out.println("Example 3: ");
+		System.out.println("a. call the API /search to get all Certified Product Details");
+		System.out.println("b. Save results to a .txt file\n");
+		
+		System.out.println("Example 4: ");
+		System.out.println("a. Call the API /data/search_options using token for authentication\n");
+		
+		System.out.println("NOTE: before running, please configure the environment.properties file\n");
+		
+		System.out.println("For example 1, please enter '1' into the console");
+		System.out.println("For example 2, please enter '2' into the console");
+		System.out.println("For example 3, please enter '3' into the console");
+		System.out.println("For example 4, please enter '4' into the console");
+		System.out.println("To perform all 4 examples, please enter '0' into the console");
+		System.out.println("To exit the application, please enter '9' into the console\n");
+	}
+	
+	/**
+	 * Returns a valid integer from the console
+	 * @return integer representing user input
+	 * @throws IOException for incorrect user console input
+	 */
+	public Integer getValidUserInput() throws IOException{
+		Scanner scanner = new Scanner(System.in);
+		Integer inputInt = scanner.nextInt();
+		if(inputInt.equals(null)){
+			System.out.println("User input integer is null. Setting to 0 as default.");
+			inputInt = 0;
+		}
+		else if(inputInt == 9){
```

Comment:
From a code readability standpoint, it might be nice to make some of these integers public static final (constants) with names that make it clear what they are doing.